crates/forge/tests/integration/test_state.rs

```diff
@@ -126,10 +136,215 @@ fn test_simple_syscalls() {
             "WarpChecker".to_string(),
             Path::new("tests/data/contracts/warp_checker.cairo"),
         )
-        .unwrap()
+        .unwrap(),
+        Contract::new(
+            "SequencerAddressChecker",
+            indoc!(
+                r#"
+                #[starknet::contract]
+                mod SequencerAddressChecker {
+                    use result::ResultTrait;
+                    use starknet::{ ClassHash, library_call_syscall, ContractAddress};
+
+                    #[storage]
+                    struct Storage {
+                    }
+
+                    #[external(v0)]
+                    fn get_sequencer_address(ref self: ContractState) -> ContractAddress {
+                        starknet::get_block_info().unbox().sequencer_address
+                    }
+                }
+                "#
+            )
+        )
+    );
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+fn test_library_calls() {
+    let test = test_case!(
+        indoc!(
+            r#"
+        use result::ResultTrait;
+        use starknet::{ ClassHash, library_call_syscall, ContractAddress };
+        use snforge_std::{ declare };
+
+        #[starknet::interface]
+        trait ILibraryContract<TContractState> {
+            fn get_value(
+                self: @TContractState,
+            ) -> felt252;
+
+            fn set_value(
+                ref self: TContractState,
+                number: felt252
+            );
+        }
+
+        #[test]
+        fn test_get_execution_info() {
+            let class_hash = declare('LibraryContract').class_hash;
+            let lib_dispatcher = ILibraryContractSafeLibraryDispatcher { class_hash };
+            let value = lib_dispatcher.get_value().unwrap();
+            assert(value == 0, 'Incorrect state');
+            lib_dispatcher.set_value(10);
+            let value = lib_dispatcher.get_value().unwrap();
+            assert(value == 10, 'Incorrect state');
+        }
```

Comment:
I think we should somehow verify that the library calls really take place in context of the caller state (i.e. state of the called contract doesn't change).

Reply:
> state of the called contract

what do you mean by called contract 🤔

Reply:
If you elaborate and it makes sense I will still address it in a PR fixing library calls bcs there are still some problems and I don't want to mix in to many changes here https://github.com/foundry-rs/starknet-foundry/issues/697

Reply:
discussed on slack, I wil try to come up with some test cases checking if the proper storage space is modified